int32 mode
int32 cnt
float32[2] th
